support-frontend/assets/pages/contributions-landing/components/StripePaymentRequestButton.tsx

```diff
@@ -0,0 +1,168 @@
+import { css } from '@emotion/core';
+import { Button } from '@guardian/src-button';
+import { neutral, space } from '@guardian/src-foundations';
+import { PaymentRequestButtonElement } from '@stripe/react-stripe-js';
+import React from 'react';
+import GeneralErrorMessage from 'components/generalErrorMessage/generalErrorMessage';
+import type { RenderPaymentRequestButtonInput } from 'components/StripePaymentRequestButton/StripePaymentRequestButton';
+import StripePaymentRequestButtonContainer from 'components/StripePaymentRequestButton/StripePaymentRequestButtonContainer';
+import type {
+	ContributionType,
+	OtherAmounts,
+	SelectedAmounts,
+} from 'helpers/contributions';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import { fromCountry } from 'helpers/internationalisation/countryGroup';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+
+interface StripePaymentRequestButtonProps {
+	country: IsoCountry;
+	currency: IsoCurrency;
+	isTestUser: boolean;
+	contributionType: ContributionType;
+	selectedAmounts: SelectedAmounts;
+	otherAmounts: OtherAmounts;
+}
+
+function StripePaymentRequestButton({
+	country,
+	currency,
+	isTestUser,
+	contributionType,
+	selectedAmounts,
+	otherAmounts,
+}: StripePaymentRequestButtonProps): JSX.Element {
+	return (
+		<StripePaymentRequestButtonContainer
+			country={country}
+			currency={currency}
+			isTestUser={isTestUser}
+			contributionType={contributionType}
+			selectedAmounts={selectedAmounts}
+			otherAmounts={otherAmounts}
+			renderPaymentRequestButton={({
+				type,
+				paymentRequest,
+				paymentRequestError,
+				onStripeButtonClick,
+				onCustomButtonClick,
+			}) => (
+				<PaymentRequestButton
+					country={country}
+					contributionType={contributionType}
+					type={type}
+					paymentRequest={paymentRequest}
+					paymentRequestError={paymentRequestError}
+					onStripeButtonClick={onStripeButtonClick}
+					onCustomButtonClick={onCustomButtonClick}
+				/>
+			)}
+		/>
+	);
+}
+
+// ---- Helper component ---- //
+
+interface PaymentRequestButtonProps extends RenderPaymentRequestButtonInput {
+	country: IsoCountry;
+	contributionType: ContributionType;
+}
+
+function PaymentRequestButton({
+	country,
+	contributionType,
+	type,
+	paymentRequest,
+	paymentRequestError,
+	onStripeButtonClick,
+	onCustomButtonClick,
+}: PaymentRequestButtonProps) {
+	const countryGroupId = fromCountry(country);
+
+	const isUkOrEuRecurring =
```

Comment:
perhaps time to reconsider this logic if we're testing PRB on the epic?

Reply:
Yeah definitely agree!